index.bs

```diff
@@ -13183,6 +13446,71 @@ The characteristics of a namespace object are described in [[#namespace-object]]
 </div>
 
 
+<h3 id="es-modules">Modules</h3>
+
+Modules are reified as entries in the [=module map=] as [=Synthetic Module Records=], allowing
+them to be imported as ECMAScript modules.
+
+<div algorithm>
+  To <dfn>create a synthetic module record</dfn> for a [=module=] |module| in a [=Realm=] |realm|,
+  perform the following steps:
+
+    1.  Assert: |module| is [=exposed=] in |realm|.
+    1.  Let |exports| be « ».
+    1.  [=list/For each=] |member| of |module|'s [=module/module members|members=]:
+        1.  Assert: |member| is not annotated with either the [{{Exposed}}] or [{{SecureContext}}]
+            [=extended attributes=].
+        1.  Append the [=identifier=] of |member| to |exports|.
+    1.  Let |moduleRecord| be [=!=] [=CreateSyntheticModule=](|exports|, the following steps,
+        |realm|, undefined), with the following steps:
+        1.  [=list/For each=] |member| of |module|'s [=module/module members|members=]:
+            1.  Let |id| be |member|'s [=identifier=].
+            1.  If |member| is an [=interface=]:
+                1.  Let |interfaceObject| be the result of [=create an interface object|creating
+                    an interface object=] for |member| with |id| in |realm|.
+                1.  Perform [=!=] [=SetSyntheticModuleExport=](|moduleRecord|, |id|,
+                    |interfaceObject|).
+            1.  Otherwise, if |member| is an [=operation=]:
+                1.  Let |method| be the result of [=creating an operation function=] given
+                    |member|, |module|, and |realm|.
+                1.  Perform [=!=] [=SetSyntheticModuleExport=](|moduleRecord|, |id|, |method|).
+            1.  Otherwise:
+                1.  Assert: |member| is an [=attribute=] or a [=partial interface=].
```

Comment:
Seeing this here makes me think that yes, we should exclude partial interfaces. (Which is an inline issue above.)

Reply:
I don't think there's a real issue here; I think we should continue supporting partial interfaces in modules. For a use case, imagine if the Temporal proposal defined `CivilDate` in ECMA-262, in a built-in module, and then defined `CivilDate.prototype.toLocaleString` in ECMA-402 -- it would make sense to define the `toLocaleString` method in a partial interface inside a partial module. (We currently use a weirder system for Intl, where one spec has one toLocaleString definition, and the other overrides it, but I think this is even messier.)

I think we're concerned about this for the wrong reasons--this is really  more of an editorial mismatch vs the rest of the specification than an actual difficulty being faced. For example, when [defining the global property references](https://heycam.github.io/webidl/#define-the-global-property-references), there is a loop over "every non-callback interface that is exposed in realm.". That is another way of skipping partial interfaces, just less imperatively. I think the eventual fix to #432 will make it even more obvious that things work the same way between these two cases.

Reply:
Oh, for sure, we should support partial interfaces inside modules. But should they really be in the member list? Or should only the primary interfaces be in the member list, with the partial interfaces just merging their interface-members into the primary interface?

Reply:
The main reason I left them in the member list is to formally define "enclosing module" for partial interfaces. I agree it's a little awkward, though.

Reply:
Thanks for clarifying, you two. I don't have an opinion on this phrasing.